naturblick/views/ObservationListView.swift

```diff
@@ -45,6 +54,53 @@ class ObservationListViewController: HostingController<ObservationListView> {
            menuVC.popoverPresentationController?.barButtonItem = sender as? UIBarButtonItem
            navigationController?.present(menuVC, animated: true)
     }
+    
+    func configureNavigationItem() {
+        if (model.selectedItems.isEmpty) {
+            let addButton = UIBarButtonItem(image: UIImage(named: "add_24"), style: .plain, target: self, action: #selector(ObservationListViewController.openMenu))
+            let mapButton = UIBarButtonItem(primaryAction: UIAction(image: UIImage(named: model.showList ? "map" : "format_list_bulleted")) {action in
+                self.model.showList.toggle()
+            })
+            addButton.accessibilityLabel = String(localized: "acc_add")
+            mapButton.accessibilityLabel = String(localized: model.showList ? "acc_map" : "acc_format_list_bulleted")
+            navigationItem.setRightBarButtonItems( [ addButton, mapButton ], animated: true)
+        } else {
+            let deleteButton = UIBarButtonItem(image: UIImage(named: "trash_24"), style: .plain, target: self, action: #selector(ObservationListViewController.deleteObservations))
+            deleteButton.accessibilityLabel = String(localized: "acc_delete")
+            deleteButton.tintColor = UIColor(Color.onPrimaryHighEmphasis)
+            navigationItem.setRightBarButtonItems( [ deleteButton ], animated: true)
+        }
+    }
+    
+    @objc func deleteObservations(_ sender: Any?) {
+        var title: String
+        var msg: String
+        
+        if (model.selectedItems.count == 1) {
+            title = String(localized: "delete_question")
+            msg = String(localized: "delete_question_message")
+        } else {
+            let countStr = "\(model.selectedItems.count)"
+            title = String(localized: "delete_question_plural")
+            msg = String(localized: "delete_question_message_plural \(countStr)")
+        }
+        
+        let alert = UIAlertController(title: title, message: msg, preferredStyle: .actionSheet)
+        alert.addAction(UIAlertAction(title: String(localized: "delete"), style: .destructive, handler: { _ in
+            do {
+                for obs in self.model.selectedItems {
```

Comment:
I think you should enqueue all delete operations and then do refresh the DB, this refreshes it after each delete.